src/cmd-build

```diff
@@ -263,12 +276,30 @@ fi
 
 # And add the commit metadata itself, which includes notably the rpmdb pkglist
 # in a format that'd be easy to generate diffs out of for higher level tools
-"${dn}"/commitmeta_to_json "${workdir}/repo" "${commit}" > commitmeta.json
+"${dn}"/commitmeta_to_json "${tmprepo}" "${commit}" > commitmeta.json
+
+# And create a tarball of an ostree repo containing the commit
+if [ "${commit}" == "${previous_commit}" ]; then
+    cp -a --reflink=auto "${previous_builddir}/ostree-commit.tar" .
+else
+    ostree init --repo=repo --mode=archive
+    ostree pull-local --repo=repo "${tmprepo}" "${ref}"
+    if [ -n "${ref_is_temp}" ]; then
+        ostree refs --repo=repo --delete "${ref}"
+    fi
+    # Don't compress; archive repos are already compressed individually and we'd
+    # gain ~20M at best. We could probably have better gains if we compress the
+    # whole repo in bare/bare-user mode, but that's a different story...
+    tar -cf ostree-commit.tar -C repo .
```

Comment:
Simple enough.  But one downside of this is that today libostree doesn't know how to pull "tarball of ostree repo" and I can see wanting that.  We could teach it...wouldn't be too hard.  But maybe worth considering using a static delta?

Actually in general I think we should make this a config option as I'm not sure we necessarily want it for RHCOS since we have the oscontainers.  (Maybe.  Data is cheap assuming we have GC)

(And making it a config option would raise the question of a config file)

Reply:
> Actually in general I think we should make this a config option as I'm not sure we necessarily want it for RHCOS since we have the oscontainers.

Hmm, how about a `--delete-ostree-repo` flag to `oscontainer` to delete the repo and just leave the `ostree-commit-object` behind once the container is successfully built?

Reply:
We could also just leave the repo directory, but make the commit partial so we just keep the same logic when searching for the commit object.

Reply:
So now that we do keep the `ostree-commit-object` again, let me reword and amend my previous proposal: since `oscontainer` produces something very similar already (a tarball containing the repo), how about (1) we make it `${workdir}` aware so rather than passing a repo path, we pass it a build id (or default to latest) like other commands (see also [this comment](https://github.com/coreos/coreos-assembler/pull/515#issuecomment-493193392) about this), and (2) give it a `--consume` flag which deletes the tarball once it's done?